feat(mme): 5G Traffic test related fixes

panyogesh opened the pull request:
Fix :
  1. Changed the local IP Address recieved from sessiond to network
     format and relaying to GNB.
  2. Changing the default bandwidth to 1 Gigabits

Test:
  1. Tested using UERANSIM. Packets are reaching table=0

cookie=0x0, duration=224.372s, table=0, n_packets=17, n_bytes=1428
cookie=0x0, duration=224.354s, table=0, n_packets=17, n_bytes=1666

Signed-off-by: Yogesh Pandey <yogesh@wavelabs.ai>

<!--
    Tag your PR title with the components that it touches along with
    the type of change
    E.g. "fix(orc8r): Fix reindexer race condition" or "feat(agw) ..."
-->

## Summary

<!-- Enumerate changes you made and why you made them -->

## Test Plan

<!--
    How did you test your change? How do you know it works?
    Add supporting screenshots, terminal pastes, etc. as necessary
-->

## Additional Information

- [ ] This change is backwards-breaking

<!--
    If this is a backwards-breaking change, document the upgrade instructions.
    All upgrade instructions for backwards-breaking changes will be aggregated
    in the next release's changelog so this is very important to fill out.
-->

panyogesh committed:
feat(mme): 5G Traffic test related fixes

Fix :
  1. Changed the local IP Address recieved from sessiond to network
     format and relaying to GNB.
  2. Changing the default bandwidth to 1 Gigabits

Test:
  1. Tested using UERANSIM. Packets are reaching table=0

Signed-off-by: Yogesh Pandey <yogesh@wavelabs.ai>

panyogesh commented:
[Test-Logs-8583.txt](https://github.com/magma/magma/files/6963669/Test-Logs-8583.txt)

uri200 commented on lte/protos/session_manager.proto:
```diff
@@ -935,11 +935,36 @@ message RatSpecificContextAccess {
+   Pbps_4   = 22;
+   Pbps_16  = 23;
+   Pbps_64  = 24;
+   Pbps_256 = 25;
```
Instead of defining every single constant like on 3gpp, can't we just pass the raw value and do the conversion at `cpp` level if required?

I have the feeling we are defining three times these constants just to convert them back and forth across the AMF, SMF, Pipelined. I think the only place it really matters to know what those value mean is on the system applying the Qos (I guess pipelined). For the rest of the system, the raw value is enough. I don't think thre is any difference for sessiond to use `3` int vs `Kbps_16` enum

This way we keep the logic in the services. Proto shouldn't really care what the value means. Plus may force the sender to do an extra conversion from raw value into an enum



The system that needs to understand what raw values need it will have to have a function that matches raw to a actual value: 
```
0 to int(0)
1 to int(1)
3 to int(16)
10 to int(256000000)
...
```

panyogesh replied:
Comments addressed.

uri200 commented on lte/gateway/c/session_manager/SessionStateEnforcer.cpp:
```diff
@@ -576,12 +576,15 @@ void SessionStateEnforcer::prepare_response_to_access(
   rsp->mutable_downlink_unit_type()->set_ambr_unit_type(
-      magma::AmbrUnit::Kbps_64);
-  rsp->mutable_uplink_unit_type()->set_ambr_unit_type(magma::AmbrUnit::Kbps_64);
+      magma::AmbrUnit::Gbps_1);
```
see my comment at the [bottom](https://github.com/magma/magma/pull/8583/files#r686339624) first:

in this case if we want to set any ambrunit we will need to create a switch function that transforms the  QOS value (for example 11) into `magma::AmbrUnit::Gbps_1`.  If you don't define any enum you can just do 

```
 rsp->mutable_downlink_unit_type()->set_ambr_unit_type(config.rat_specific_context.m5gsm_session_context().qos.get_ambr_unit_type());

```
(note qos access through config may be wrong in my example)

panyogesh replied:
Right now for 5G context,  sessiond is using a default value. As part of QoS changes this value will be derived from either default subscriber profile or values stored in policydb.  The qos infrastructure like "qos.get_ambr_unit_type" still needs to be updated for 5G Context in sessiond.   Have tried to keep the changes in line based on the comments.

panyogesh committed:
feat(amf): Traffic test related fixes

Fix:
  Traffic Test related changes.
  1. Fixed the IP Address recevied from SMF
  2. Fixed the bandwidth related ambr issues.

Test:
  1. Tested on Simulator
  2. Tested on DSTester

Signed-off-by: Yogesh Pandey <yogesh@wavelabs.ai>

uri200 commented on lte/protos/session_manager.proto:
```diff
@@ -933,17 +933,6 @@ message RatSpecificContextAccess {
-  }
-  AmbrUnitType ambr_unit_type = 1;
-}
-
```
removing this looks good. 

For future cases, remember we can pass around the raw value. Let the service that uses that value to decode it without the need to go through 3gpp structures

uri200 commented on lte/gateway/c/session_manager/SessionStateEnforcer.cpp:
```diff
@@ -576,12 +576,11 @@ void SessionStateEnforcer::prepare_response_to_access(
+  rsp->mutable_session_ambr()->set_br_unit(AggregatedMaximumBitrate::KBPS);
+  rsp->mutable_session_ambr()->set_max_bandwidth_ul(DEFAULT_AMBR_UNITS);
+  rsp->mutable_session_ambr()->set_max_bandwidth_dl(DEFAULT_AMBR_UNITS);
+
```
can you please point me to any unit test that tests this function `prepare_response_to_access`

panyogesh replied:
prepare_response_to_access : No direct UT for it.  Extended coverage for m5g_move_to_active_state() which calls the function internally.

ulaskozat commented:
Based on my conversation with @panyogesh, session AMBR is temporarily hardwired in sessiond. Since this is the same as APN AMBR in 4G, we will reuse APN profile and AMF will get the value and will fetch it to sessiond in a separate PR.

panyogesh committed:
feat(amf): Traffic test related fixes

Fix:
  Traffic Test related changes.
  1. Fixed the IP Address recevied from SMF
  2. Fixed the bandwidth related ambr issues to  1mbps.

Test:
  1. Tested on Simulator
  2. Tested on DSTester

Signed-off-by: Yogesh Pandey <yogesh@wavelabs.ai>

panyogesh commented:
Test Logs
---------------
vagrant@ue-ran-sim:~$ ping 192.168.129.42 -I  192.168.128.254 -c 5
PING 192.168.129.42 (192.168.129.42) from 192.168.128.254 : 56(84) bytes of data.
64 bytes from 192.168.129.42: icmp_seq=1 ttl=63 time=5.92 ms


      Start 17: test_set_session_manager_handler
17/17 Test #17: test_set_session_manager_handler .....   Passed    0.23 sec

ulaskozat reviewed (approved):
LGTM

ulaskozat commented:
@uri200 could you approve if your concerns are addressed?
